src/monster.cpp

```diff
@@ -2412,7 +2423,7 @@ void monster::process_one_effect( effect &it, bool is_new )
     int val = get_effect( "HURT", reduced );
     if( val > 0 ) {
         if( is_new || it.activated( calendar::turn, "HURT", val, reduced, 1 ) ) {
-            apply_damage( nullptr, bodypart_id( "torso" ), val );
+            apply_damage( source_creature, bodypart_id( "torso" ), val );
```

Comment:
I think `apply_damage` should probably change to taking an `effect_source` argument, but that probably shouldn't be part of this PR.